plugins/MantisCoreFormatting/tests/MarkdownTest.php

```diff
@@ -14,79 +14,395 @@
+	 * - "true" => Parsdown::text(); Paragraph is expected.
+	 */
+	public function testConvertToLineOrParagraph(): void {
+		$t_sample = 'lorem ipsum dolor';
+
+		$this->assertSame( $t_sample, $this->parser->convert( $t_sample ) );
+		$this->assertSame( '<p>' . $t_sample . '</p>', $this->parser->convert( $t_sample, true ) );
+	}
+
+	public function provideHeaders(): Generator {
+		# valid headers
+		yield  'valid: # foo' => [ '# foo', '<h1>foo</h1>' ];
+		yield  'valid: ## foo' => [ '## foo', '<h2>foo</h2>' ];
+		yield  'valid: ### foo' => [ '### foo', '<h3>foo</h3>' ];
+		yield  'valid: #### foo' => [ '#### foo', '<h4>foo</h4>' ];
+		yield  'valid: ##### foo' => [ '##### foo', '<h5>foo</h5>' ];
+		yield  'valid: ###### foo' => [ '###### foo', '<h6>foo</h6>' ];
+
+		# invalid headers
+		yield  'invalid: ####### foo' => [ '####### foo', '<p>####### foo</p>' ];
+		yield  'invalid: #foo' => [ '#foo', '<p>#foo</p>' ];
+		yield  'invalid: #123' => [ '#123', '<p>#123</p>' ];
+	}
+
+	public function provideEmails(): Generator {
+		yield 'process_urls = ON; lorem <user@exmaple.com> ipsum' => [
+			'lorem <user@exmaple.com> ipsum',
+			ON,
+			'<p>lorem <a href="mailto:user@exmaple.com">user@exmaple.com</a> ipsum</p>'
+		];
+
+		yield 'process_urls = OFF; lorem <user@exmaple.com> ipsum' => [
+			'lorem <user@exmaple.com> ipsum',
+			OFF,
+			'<p>lorem <a href="mailto:user@exmaple.com">user@exmaple.com</a> ipsum</p>'
+		];
+
+		yield 'process_urls = ON; lorem user@exmaple.com ipsum' => [
+			'lorem user@exmaple.com ipsum',
+			ON,
+			'<p>lorem <a href="mailto:user@exmaple.com">user@exmaple.com</a> ipsum</p>'
+		];
+
+		yield 'process_urls = OFF; lorem user@exmaple.com ipsum' => [
+			'lorem user@exmaple.com ipsum',
+			OFF,
+			'<p>lorem user@exmaple.com ipsum</p>'
+		];
+	}
+
+	public function provideUrls(): Generator {
+		yield 'process_urls = ON; lorem <https://exmaple.com> ipsum' => [
+			'lorem <https://exmaple.com> ipsum',
+			ON,
+			'<a href="https://exmaple.com"',
+			true
+		];
+
+		yield 'process_urls = OFF; lorem <https://exmaple.com> ipsum' => [
+			'lorem <https://exmaple.com> ipsum',
+			OFF,
+			'<a href="https://exmaple.com"',
+			true
+		];
+
+		yield 'process_urls = ON; lorem [link](https://exmaple.com) ipsum' => [
+			'lorem [link](https://exmaple.com) ipsum',
+			ON,
+			'<a href="https://exmaple.com"',
+			true
+		];
+
+		yield 'process_urls = OFF; lorem [link](https://exmaple.com) ipsum' => [
+			'lorem [link](https://exmaple.com) ipsum',
+			ON,
+			'<a href="https://exmaple.com"',
+			true
+		];
+
+		yield 'process_urls = ON; lorem https://exmaple.com ipsum' => [
+			'lorem https://exmaple.com ipsum',
+			ON,
+			'<a href="https://exmaple.com"',
+			true
+		];
+
+		yield 'process_urls = OFF; lorem https://exmaple.com ipsum' => [
+			'lorem https://exmaple.com ipsum',
+			OFF,
+			'<a href="https://exmaple.com"',
+			false
+		];
+	}
+
+	public function provideAttributes(): Generator {
+		yield 'LINKS_NEW_WINDOW' => [
```

Comment:
Isn't that redundant, already covered by the changes I recently merged (#1983) ?